rskj-core/src/test/java/co/rsk/peg/federation/FederationChangeIT.java

```diff
@@ -0,0 +1,570 @@
+            federation.getP2SHScript();
+    }
+
+    private static Transaction buildUpdateCollectionsTx() {
+        var nonce = 3;
+        var value = 0;
+        var gasPrice = BigInteger.valueOf(0);
+        var gasLimit = BigInteger.valueOf(100000);
+        var rskTx = CallTransaction.createCallTransaction(nonce, gasPrice.longValue(),
+                gasLimit.longValue(), PrecompiledContracts.BRIDGE_ADDR, value,
+                Bridge.UPDATE_COLLECTIONS, Constants.MAINNET_CHAIN_ID);
+        var randomKey = BtcECKey.fromPrivate(Hex.decode("45c5b07fc1a6f58892615b7c31dca6c96db58c4bbc538a6b8a22999aaa860c32"));
+        rskTx.sign(randomKey.getPrivKeyBytes());
+        return rskTx;
+    }
+    
+    
+    /* Assert and verify federation change related methods */
+
+    private void assertUTXOsReferenceMovedFromNewToOldFederation(List<UTXO> utxos) {
+        // Assert old federation exists in storage
+        assertNotNull(
+            federationStorageProvider.getOldFederation(BRIDGE_CONSTANTS.getFederationConstants(), ACTIVATIONS));
+        // Assert new federation exists in storage
+        assertNotNull(
+            federationStorageProvider.getNewFederation(BRIDGE_CONSTANTS.getFederationConstants(), ACTIVATIONS));
+        // Assert old federation holds the original utxos
+        List<UTXO> utxosToMigrate = federationStorageProvider.getOldFederationBtcUTXOs();
+        assertTrue(utxos.stream().allMatch(utxosToMigrate::contains));
+        // Assert the new federation does not have any utxos yet
+        assertTrue(federationStorageProvider
+            .getNewFederationBtcUTXOs(BRIDGE_CONSTANTS.getBtcParams(), ACTIVATIONS)
+            .isEmpty());
+    }
+
+    private void assertNewAndOldFederationsHaveExpectedAddress(
+          Address expectedNewFederationAddress, Address expectedOldFederationAddress) {
+        // New active and retiring federation
+        assertEquals(expectedNewFederationAddress, bridgeSupport.getActiveFederationAddress());
+        assertEquals(expectedOldFederationAddress, bridgeSupport.getRetiringFederationAddress());
+    }
+    
+    private void assertMigrationHasNotStarted() throws Exception {
+        // Current block is behind fedActivationAge + fundsMigrationAgeBegin
+        var blockNumber = BRIDGE_CONSTANTS.getFederationConstants().getFederationActivationAge(ACTIVATIONS) + 
+            BRIDGE_CONSTANTS.getFederationConstants().getFundsMigrationAgeSinceActivationBegin();
+        assertTrue(currentBlock.getNumber() <= blockNumber);
+
+        // Pegouts waiting for confirmations should be empty
+        assertTrue(bridgeStorageProvider.getPegoutsWaitingForConfirmations().getEntries().isEmpty());
+    }
+     
+    private void assertMigrationHasStarted() throws Exception {
+        // Pegouts waiting for confirmations should not be empty
+        assertFalse(bridgeStorageProvider.getPegoutsWaitingForConfirmations().getEntries().isEmpty());
```

Comment:
Maybe we can check that there is only one element? Should be the case as long as there are less than 50 UTXOs to migrate

Reply:
I have updated this. Made sure that the UTXO list generated at the beginning is always less than 50 to accomodate this assertion.